packages/sui-i18n/README.md

```diff
@@ -59,43 +60,134 @@ i18n.t('PLURAL_WITH_VARIABLE', {houses: 2}) //=> 2 casas
 i18n.n(1000) //=> 1.000
 i18n.n(1000000) //=> 1.000.000
 i18n.f('phone', '123123123') //=> '123 123 123'
-
 ```
 
 From now on, the library will use Polyglot to translate your literals anywhere in your app.
 
+### I18n.prototype.t vs I18n.prototype.interpolate
+
+By default, the `I18n.prototype.t` method should always be your first choice when retrieving translations. It already comes with many features such as interpolating placeholders with static values.
+
+```js
+i18n.t('HELLO_WORLD') //=> ¡Hola mundo!
+i18n.t('PLURAL', 1) //=> uno
+i18n.t('PLURAL', 2) //=> varios
+i18n.t('PLURAL_WITH_VARIABLE', {houses: 1}) //=> 1 casa
+i18n.t('PLURAL_WITH_VARIABLE', {houses: 2}) //=> 2 casas
+```
+
+However, in some exceptional cases we may want to parse our translations with a stronger interpolation algorithm. For instance, we may want to collect some text between specific boundaries, and we want to return a specific result for that specific chunk of text.
+
+A typical use case is when a literal contains special styles or transformation in a React component, like an anchor tag in the middle of the text or anything else that would require enhance a portion of text.
+
+`I18n.prototype.interpolate` has all the features of `I18n.prototype.t` and provide the additional flexibility of collecting the children content between boundaries, so that we can return the desired value.
+
+It **DOES NOT** return a string anymore, instead it'll give back an array of chunks.
+
+The placeholder for interpolating boundaries is different from the one for static values, you'll need to wrap your content between `%[key]children[key]%`, for example:
+
+```text
+SIMPLE_BOUNDARY: 'Your name is %[upper]Spike[upper]%, the best superhero!'
+```
+
+The parser supports:
+
+- Multiple sequential groups.
+- Nested interpolation
+
+```javascript
+import I18n from '@s-ui/i18n'
+import Polyglot from '@s-ui/i18n/lib/adapters/polyglot'
+
+const i18n = new I18n({adapter: new Polyglot()})
+i18n.languages = {
+  'en-GB': {
+    SIMPLE_BOUNDARY: 'Your name is %[upper]Spike[upper]%, the best superhero!',
+    SIMPLE_BOUNDARY_PLACEHOLDER:
+      '%{name}, your name is %[upper]%{name}[upper]%!',
+    MULTIPLE_BOUNDARY:
+      '%[upper]%{name}[upper]%, your name is %[upper]%{name}[upper]%!',
+    NESTED_PLACEHOLDERS:
+      'Hello, this is an %[bold]important %[link]resource[link]%[bold]% for this page!'
+  }
+}
+
+i18n.culture = 'es-ES'
+i18n.interpolate('SIMPLE_BOUNDARY', {
+  name: 'Spike',
+  upper: ({children}) => children.toUpperCase()
+}) //=> ['Spike, your name is ', 'SPIKE', '!']
+i18n.interpolate('SIMPLE_BOUNDARY_PLACEHOLDER', {
+  name: 'Spike',
+  upper: ({children}) => children.toUpperCase()
+}) //=> ['Spike, your name is ', 'SPIKE', '!']
+i18n.interpolate('MULTIPLE_BOUNDARY', {
+  name: 'Spike',
+  upper: ({children}) => children.toUpperCase()
+}) //=> ['SPIKE', ', your name is ', 'SPIKE', '!']
```

Comment:
Could you fix this comment with correct value? Also, line 123 is not correct

Reply:
Mmm I don't see the error, `SIMPLE_BOUNDARY_PLACEHOLDER` and `MULTIPLE_BOUNDARY` are different from the `SIMPLE_BOUNDARY` copy